huaweicloud/services/cce/resource_huaweicloud_cce_cluster_log_config.go

```diff
@@ -0,0 +1,260 @@
+package cce
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var logConfigNonUpdatableParams = []string{"cluster_id"}
+
+// @API CCE PUT /api/v3/projects/{project_id}/cluster/{cluster_id}/log-configs
+// @API CCE GET /api/v3/projects/{project_id}/cluster/{cluster_id}/log-configs
+func ResourceClusterLogConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceClusterLogConfigCreateOrUpdate,
+		UpdateContext: resourceClusterLogConfigCreateOrUpdate,
+		ReadContext:   resourceClusterLogConfigRead,
+		DeleteContext: resourceClusterLogConfigDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceClusterLogConfigImport,
+		},
+
+		CustomizeDiff: config.FlexibleForceNew(logConfigNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ttl_in_days": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"log_configs": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"enable": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func buildClusterLogConfigCreateBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"ttl_in_days": utils.ValueIgnoreEmpty(d.Get("ttl_in_days")),
+		"log_configs": buildLogConfigsCreateBodyParams(d),
+	}
+	return bodyParams
+}
+
+func buildLogConfigsCreateBodyParams(d *schema.ResourceData) []map[string]interface{} {
+	logConfigsRaw := d.Get("log_configs").(*schema.Set).List()
+	if len(logConfigsRaw) == 0 || logConfigsRaw[0] == nil {
+		return nil
+	}
+
+	bodyParams := make([]map[string]interface{}, len(logConfigsRaw))
+	for i, v := range logConfigsRaw {
+		logConfig := v.(map[string]interface{})
```

Comment:
if name and enable are both empty, then a panic will be occured

Reply:
fixed